pbsmetrics/go_metrics_test.go

```diff
@@ -99,6 +102,29 @@ func ensureContainsAdapterMetrics(t *testing.T, registry metrics.Registry, name
 	ensureContainsBidTypeMetrics(t, registry, name, adapterMetrics.MarkupMetrics)
 }
 
+func TestDisableAccountMetrics(t *testing.T) {
+	accountId := "acct-id"
+	registry := metrics.NewRegistry()
+	m := NewMetrics(registry, []openrtb_ext.BidderName{openrtb_ext.BidderAppnexus}, config.DisabledMetrics{AccountAdapterDetails: true})
+
+	m.RecordAdapterBidReceived(AdapterLabels{
+		Adapter: openrtb_ext.BidderAppnexus,
+		PubID:   accountId,
+	}, openrtb_ext.BidTypeBanner, true)
+
+	// Assert no adapter metrics are found in the accountMetrics map
+	assert.Equalf(t, len(m.accountMetrics[accountId].adapterMetrics) == 0, true, "Test failed. Account metrics that contain adapter information are disabled, therefore we expect no entries in m.accountMetrics[accountId].adapterMetrics, we have %d \n", len(m.accountMetrics[accountId].adapterMetrics))
+
+	// Assert no account metrics exist in the registry
+	registryMap := registry.GetAll()
```

Comment:
We should break this up into two different tests because right now we're testing two things here: 1. When we call `NewMetrics` with a account-adapter metrics being disabled, the registry shouldn't have any account-adapter metrics (L119-125) and 2. When we call `RecordAdapterBidReceived`, there shouldn't be any account-adapter metrics recorded (because they shouldn't exist to start with) (L110 - L116)

Also, this file has a `TestRecordBidType` method that tests `RecordAdapterBidReceived`, so we should just add `#2` from above to that method.

Reply:
Done